Fix (from upstream) for drive and host endianess mismatch. See,

http://trac.xiph.org/browser/trunk/cdparanoia (upstream revision 15356)
http://bugs.gentoo.org/240960 (downstream bug)

diff -ur cdparanoia-III-10.2.orig/interface/interface.c cdparanoia-III-10.2/interface/interface.c
--- cdparanoia-III-10.2.orig/interface/interface.c	2008-09-11 13:43:52.000000000 +0300
+++ cdparanoia-III-10.2/interface/interface.c	2009-06-20 20:57:03.000000000 +0300
@@ -118,7 +118,7 @@
 	if(d->bigendianp==-1) /* not determined yet */
 	  d->bigendianp=data_bigendianp(d);
 	
-	if(d->bigendianp!=bigendianp()){
+	if(buffer && d->bigendianp!=bigendianp()){
 	  int i;
 	  u_int16_t *p=(u_int16_t *)buffer;
 	  long els=sectors*CD_FRAMESIZE_RAW/2;
